Automattic/wp-calypso
# Migration: Import user list filtering

Fixes https://github.com/Automattic/wp-calypso/issues/88502

## Proposed Changes

* Add functionality to filter by role or search query
<img width="791" alt="CleanShot 2024-03-13 at 23 50 16@2x" src="https://github.com/Automattic/wp-calypso/assets/10482592/90721339-ff62-410a-880e-dc2957a0f22f">


## Testing Instructions


* Checkout branch or use live link.
* Navigate to `/sites?hosting-flow=true` and click `Migrate a site`.
* Go through the migration flow until you reach the import user screen.
* Verify you can filter by searching or user roles. Try to break it with different filter combinations.
* Verify the `select all users` toggle and invitations are still working.

## Pre-merge Checklist
